src/Storages/StorageDeltaLake.cpp

```diff
@@ -57,20 +56,22 @@ std::vector<String> DeltaLakeMetadata::listCurrentFiles() &&
     return keys;
 }
 
-JsonMetadataGetter::JsonMetadataGetter(const StorageS3::Configuration & configuration_, ContextPtr context)
+template <typename Configuration, typename MetaReadHelper>
```

Comment:
I'd prefer it was done with inheritance instead of templates, but templates are ok, no need to change. But when remote storages are rewritten to use the interface of IObjectStorage (from Disks/IObjectStorage) this code will probably rewritten. But not soon, so ok for now with templates.